src/fireblocks-sdk.ts

```diff
@@ -1868,61 +1912,90 @@ export class FireblocksSDK {
+     *
+     * @returns WriteCallFunctionResponseDto
      */
     public async writeContractCallFunction(blockchainId: string, contractAddress: string, payload: WriteCallFunctionDto): Promise<WriteCallFunctionResponseDto> {
         return await this.apiClient.issuePostRequest(`/v1/contract-service/contract/${blockchainId}/${contractAddress}/function/write`, payload);
     }
 
     /**
-     * Issue a new token and link it to the tenant
-     * @param payload
+     * Creates a new asset and links it as a token.
+     *
+     * @param {IssueTokenRequest} payload - The payload containing information for token issuance.
+     *
+     * @returns {Promise<CreateTokenResponseDto>} Response with created token ID
      */
-    public async issueNewToken(payload: IssueTokenRequest): Promise<PendingTokenLinkDto> {
+    public async issueNewToken(payload: IssueTokenRequest): Promise<CreateTokenResponseDto> {
         return await this.apiClient.issuePostRequest(`/v1/tokenization/tokens`, payload);
     }
 
     /**
-     * Get all tokens linked to the tenant
-     * @param pageSize
-     * @param pageCursor
+     * Retrieves all linked tokens in a paginated format.
+     *
+     * @param {Object} payload - The payload for retrieving linked tokens
+     * @param {TokenLinkStatus} payload.status - The status of linked tokens (COMPLETED / PENDING)
+     * @param {number} payload.pageSize - The number of results to return on the next page
+     * @param {string} payload.pageCursor - The cursor for the next page
+     *
+     * @returns {Promise<Web3PagedResponse<TokenLink>>} A paginated array of linked tokens
      */
-    public async getLinkedTokens(pageSize: number = 100, pageCursor?: string): Promise<Web3PagedResponse<TokenLink>> {
-        const requestFilter = {
+    public async getLinkedTokens({ status = TokenLinkStatus.COMPLETED, pageSize = 100, pageCursor }: GetTokenLinksFilter = {}): Promise<Web3PagedResponse<TokenLink>> {
+        return await this.apiClient.issueGetRequest(`/v1/tokenization/tokens?${queryString.stringify({
+            status,
             pageSize,
-            pageCursor
-        };
-        return await this.apiClient.issueGetRequest(`/v1/tokenization/tokens?${queryString.stringify(requestFilter)}`);
+            pageCursor,
+        })}`);
+    }
+
+    /**
+     * Get the total count of linked tokens
+     * @returns TokenLinksCount
+     */
+    public async getLinkedTokensCount(): Promise<TokenLinksCount> {
+        return await this.apiClient.issueGetRequest(`/v1/tokenization/tokens/count`);
     }
 
     /**
      * Link a token to the tenant
-     * @param assetId
+     * @param type
+     * @param refId
+     *
+     * @returns TokenLink
      */
-    public async linkToken(assetId: string): Promise<TokenLink> {
-        return await this.apiClient.issuePutRequest(`/v1/tokenization/tokens/${assetId}/link`, {});
+    public async linkToken(type: SupportedContractTemplateType, refId: string): Promise<TokenLink> {
```

Comment:
you are using just `id` in other endpoints (not `refId`)

Reply:
as we talked, `id` is being generated only after the token link was created/linked and not before